From 52adbbe3d6c7cd17ca14bdf5b1337348bb5c03e9 Mon Sep 17 00:00:00 2001
From: Matt DeVillier <matt.devillier@gmail.com>
Date: Fri, 25 Apr 2025 13:15:47 -0500
Subject: [PATCH 02/25] MdeModulePkg: Don't remove rejected PCI devices

This is needed to fix framebuffer display when using
`CONFIG_RESOURCE_ALLOCATION_TOP_DOWN=y` in coreboot. Without this
patch, the display backlight will be on, but no boot logo or
menus visible.

The code detecting valid BARs is buggy AF, so just skip the whole
mess for now.

Signed-off-by: Matt DeVillier <matt.devillier@gmail.com>
(cherry picked from commit 7343ef1d0f08689434eb52724246a450cdf9347e)
---
 MdeModulePkg/Bus/Pci/PciBusDxe/PciEnumeratorSupport.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/MdeModulePkg/Bus/Pci/PciBusDxe/PciEnumeratorSupport.c b/MdeModulePkg/Bus/Pci/PciBusDxe/PciEnumeratorSupport.c
index ae94462f60..a6ff9e8ab3 100644
--- a/MdeModulePkg/Bus/Pci/PciBusDxe/PciEnumeratorSupport.c
+++ b/MdeModulePkg/Bus/Pci/PciBusDxe/PciEnumeratorSupport.c
@@ -2883,6 +2883,9 @@ IsPciDeviceRejected (
     return FALSE;
   }
 
+  // This code is buggy AF, so don't reject anything
+  return FALSE;
+
   if (IS_CARDBUS_BRIDGE (&PciIoDevice->Pci)) {
     //
     // Only test base registers for P2C
-- 
2.55.0

